packages/cli/src/local-data/replay-assets.ts

```diff
@@ -36,29 +43,26 @@ export const loadAssetMetadata: () => Promise<AssetMetadata> = async () => {
 export const saveAssetMetadata: (
   assetMetadata: AssetMetadata
 ) => Promise<void> = async (assetMetadata) => {
-  const assetsDir = join(getMeticulousLocalDataDir(), "assets");
-  await mkdir(assetsDir, { recursive: true });
-  const assetsFile = join(assetsDir, `assets.json`);
+  const assetsFile = join(await getCreateAssetsDir(), ASSET_METADATA_FILE_NAME);
 
   await writeFile(assetsFile, JSON.stringify(assetMetadata, null, 2));
 };
 
 export const fetchAsset: (path: string) => Promise<string> = async (path) => {
   const logger = log.getLogger(METICULOUS_LOGGER_NAME);
   const fetchUrl = new URL(path, getSnippetsBaseUrl()).href;
-
-  const assetsDir = join(getMeticulousLocalDataDir(), "assets");
+  const assetFileName = basename(new URL(fetchUrl).pathname);
 
   const assetMetadata = await loadAssetMetadata();
-  const etag = (await axios.head(fetchUrl)).headers["etag"] || "";
+  const etag = (await axios.head(fetchUrl)).headers["etag"] || null;
```

Comment:
Is there anything against doing this?
```suggestion
  const etag = (await axios.head(fetchUrl)).headers["etag"] || "";
```

Then the type can remain `string`.

Reply:
I feel like `null` provides more clarity that the eTag wasn't set rather than an empty string (which might also be a valid value?)

Reply:
I don't think we should treat `""` as a valid `etag`, it does not make sense